pyvxl/pydbc.py

```diff
@@ -397,69 +145,75 @@ def t_string(self, t):
-    def t_newline(self, t):
+    def t_newline(self, t):  # noqa
         r'\n+'
         t.lexer.lineno += t.value.count('\n')
 
-class DBCParser(object):
-    """The main DBC parser object"""
-    def __init__(self, lexer=None, **kwargs):
-        self.dbc = DBCFile()
-        self.signalDict = {}
-        self.signalDictByName = {}
-        self.send_types = []
-        self.nodeDict = {}
-        self.msgDict = {}
-        if lexer:
-            self.lexer = lexer
-        else:
-            self.lexer = DBCLexer()
 
-        self.tokens = self.lexer.tokens
-        self.parser = yacc.yacc(module=self, **kwargs)
-
-    def parse(self, dbc):
-        """Runs the dbc through the lexer"""
-        if dbc:
-            return self.parser.parse(dbc, self.lexer.lexer, 0, 0, None)
-        else:
-            return []
+class DBCParser:
+    """A yacc based DBC parser."""
 
+    def __init__(self, path, node_type, message_type, signal_type, **kwargs):  # noqa
+        self.node_type = node_type
+        self.message_type = message_type
+        self.signal_type = signal_type
+        self.nodes = {}
+        self.messages = {}
+        self.signals = {}
+        self.send_types = []
+        self.send_types_expected = ['cyclicX', 'spontanX', 'cyclicIfActiveX',
+                                    'spontanWithDelay', 'cyclicAndSpontanX',
+                                    'cyclicAndSpontanWithDelay',
+                                    'spontanWithRepitition',
+                                    'cyclicIfActiveAndSpontanWD',
+                                    'cyclicIfActiveFast',
+                                    'cyclicWithRepeatOnDemand', 'none']
+        self.lexer = DBCLexer(debug=False)
+        self.tokens = self.lexer.tokens
+        self.parser = yacc(module=self, **kwargs)
+        with open(path, 'r') as f:
+            dbc = f.read()
+        self.parser.parse(dbc, self.lexer.lexer, 0, 0, None)
+        if not self.send_types:
+            raise AssertionError('')
+        update = True
+        if len(self.send_types) == len(self.send_types_expected):
+            for x in range(len(self.send_types)):
+                if self.send_types[x] != self.send_types_expected[x]:
+                    break
+            else:
+                update = False
 
-    # Parsing rules
+        if update:
+            for msg in self.messages:
+                msg.send_type = self.send_types[msg.send_type_num]
 
     def p_error(self, p):
-        """Prints a parsing error"""
-        print("Syntax error at token %s (%s) on line %d" % (p.type, p.value,
-                                                            p.lineno))
-        #yacc.errok()
+        """Print a parsing error."""
+        print('Syntax error at token {p.type} ({p.value}) on line {p.lineno}')
 
-    # DBC file
-    #########################################################################
-
-    ###
-    def p_dbc(self, p):
+    def p_dbc(self, p):  # noqa

```

Comment:
Can p_dbc(self, p) be removed now? I can't find it referenced anywhere.

Reply:
Yep